### PR TITLE
Etna 797

### PR DESCRIPTION
**Pull request by alex-hse-repository**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna-ts/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna-ts/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Proposed Changes

Add Clustering

### CODE REVIEW

**Comment by alex-hse-repository** [comment]
Note: The DistanceMatrix version for new architecture shows the same time as the old one in computing distance matrix

**Review by martins0n — CHANGES_REQUESTED** [review]
- We should add basic example to DTWClustering and EuclideanClustering class docstrings.
- np.array to np.ndarray in type
- check docsting for proper `Result` section. Sometimes typing is missing.

File: etna/clustering/distances/distance_matrix.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    def _compute_dist(self, series: List[pd.Series], idx: int) -> np.ndarray:
+        """Compute distance from idx-th series to other ones."""
+        distances = np.array([self.distance(series[idx], series[j]) for j in range(self.series_number)])
+        return distances
+
+    def _compute_dist_matrix(self, series: List[pd.Series]) -> np.ndarray:
+        """Compute distance matrix for given series."""
+        distances = np.empty(shape=(self.series_number, self.series_number))
+        logging_freq = self.series_number // 10
+        logger_id = tslogger.add(ConsoleLogger())
```
Comment (martins0n) [inline etna/clustering/distances/distance_matrix.py:71]: You shouldn't do this. Tslogger is as global object and all handlers've added already.
`        tslogger.start_experiment()` and this look at code examples https://github.com/tinkoff-ai/etna-ts/blob/51afbc2df6d046e3c2ab9b102f50fc7c83529f5a/etna/datasets/tsdataset.py#L122

**Review by martins0n — CHANGES_REQUESTED** [review]

**Review by martins0n — APPROVED** [review]
👍

**Pull request merged**